src/common/cluster-store.ts

```diff
@@ -225,6 +225,12 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
     workspaceStore.setLastActiveClusterId(clusterId);
   }
 
+  deactivate(id: ClusterId) {
+    if (this.isActive(id)) {
+      this.setActive(null);
+    }
+  }
```

Comment:
Note that once #2355 is merged, this will have to change. But we will resolve that race condition in the future.

Reply:
I'll get familiar with the changes in the meantime.